cmd/asset-syncer/utils.go

```diff
@@ -769,25 +798,29 @@ func parseRepoIndex(body []byte) (*helmrepo.IndexFile, error) {
 	return &index, nil
 }
 
-func chartsFromIndex(index *helmrepo.IndexFile, r *models.Repo) []models.Chart {
+func chartsFromIndex(index *helmrepo.IndexFile, r *models.Repo, shallow bool) []models.Chart {
 	var charts []models.Chart
 	for _, entry := range index.Entries {
 		if entry[0].GetDeprecated() {
 			log.WithFields(log.Fields{"name": entry[0].GetName()}).Info("skipping deprecated chart")
 			continue
 		}
-		charts = append(charts, newChart(entry, r))
+		charts = append(charts, newChart(entry, r, shallow))
 	}
 	sort.Slice(charts, func(i, j int) bool { return charts[i].ID < charts[j].ID })
 	return charts
 }
 
 // Takes an entry from the index and constructs a database representation of the
 // object.
-func newChart(entry helmrepo.ChartVersions, r *models.Repo) models.Chart {
+func newChart(entry helmrepo.ChartVersions, r *models.Repo, shallow bool) models.Chart {
 	var c models.Chart
 	copier.Copy(&c, entry[0])
-	copier.Copy(&c.ChartVersions, entry)
+	if shallow {
+		copier.Copy(&c.ChartVersions, []helmrepo.ChartVersion{*entry[0]})
```

Comment:
Both the `entry[0]` above and here `*entry[0]` look like they should be checked. but it's been like this for a while without issue (that we've noticed) so... not worth your time right now I assume :)

Reply:
:shrug: